### PR TITLE
Add featureflags to content app (dynamic) endpoints

### PR DESCRIPTION
**Pull request by chrisdecenzo**
#### Problem
* Dynamic endpoints on tv-app have feature flags empty (set to 0)
* https://github.com/project-chip/connectedhomeip/issues/19948

#### Change overview
Add ability for delegates used by tv-app to set feature flags

TODO: still need to make the same change to other media clusters

#### Testing
* tested using tv-app and tv-casting-app

### CODE REVIEW

File: examples/tv-app/linux/include/content-launcher/ContentLauncherManager.h [hunk 1]
```diff
@@ -51,11 +51,15 @@ class ContentLauncherManager : public ContentLauncherDelegate
...
+
 protected:
     std::list<std::string> mAcceptHeaderList;
     uint32_t mSupportedStreamingProtocols;
     std::vector<ContentEntry> mContentList;
 
 private:
     EndpointId mEndpointId;
+    // TODO: set this based upon meta data from app
+    uint32_t mDynamicEndpointFeatureMap = 3;
```
Comment (amitnj) [inline examples/tv-app/linux/include/content-launcher/ContentLauncherManager.h:64]: So if this is modeled like a attribute read, I think we may have an opportunity to simplify this by sending this directly to the app as with other attributes. See PR [19924](https://github.com/project-chip/connectedhomeip/pull/19924) for this. In fact we could then do this for even the supported commands & attributes and wont have to get that info via manifest.
The other option is to get this info (as is in the manifest) when adding the content app and then caching it in the native layer. advantage of that is we dont have to send intent to the app which will then defer creating the app process till an actual command is received.

Comment (amitnj) [reply examples/tv-app/linux/include/content-launcher/ContentLauncherManager.h:64]: Note that I am giving example from android platform but similar options may apply for linux as well.

File: src/app/clusters/content-launch-server/content-launch-server.cpp [hunk 2]
```diff
@@ -154,6 +146,7 @@ class ContentLauncherAttrAccess : public app::AttributeAccessInterface
 private:
     CHIP_ERROR ReadAcceptHeaderAttribute(app::AttributeValueEncoder & aEncoder, Delegate * delegate);
     CHIP_ERROR ReadSupportedStreamingProtocolsAttribute(app::AttributeValueEncoder & aEncoder, Delegate * delegate);
+    CHIP_ERROR ReadFeatureFlagAttribute(EndpointId endpoint, app::AttributeValueEncoder & aEncoder, Delegate * delegate);
```
Comment (amitnj) [inline src/app/clusters/content-launch-server/content-launch-server.cpp:149]: Will we have to do this for commandlist & attributelist as well?

File: examples/tv-app/linux/include/content-launcher/ContentLauncherManager.h [hunk 3]
```diff
@@ -51,11 +51,15 @@ class ContentLauncherManager : public ContentLauncherDelegate
     CHIP_ERROR HandleGetAcceptHeaderList(AttributeValueEncoder & aEncoder) override;
     uint32_t HandleGetSupportedStreamingProtocols() override;
 
+    uint32_t GetFeatureMap(chip::EndpointId endpoint) override;
```
Comment (amitnj) [inline examples/tv-app/linux/include/content-launcher/ContentLauncherManager.h:54]: Wont we have to do this for android platform as well?

**Review by lazarkov — APPROVED** [review]

File: src/app/clusters/content-launch-server/content-launch-server.cpp (hunk 2 above)
Comment (chrisdecenzo) [reply src/app/clusters/content-launch-server/content-launch-server.cpp:149]: no, this part seems to work already based upon the code in AppImpl which creates the dynamic endpoint. There is just not a way to set the feature flag using those methods

File: examples/tv-app/linux/include/content-launcher/ContentLauncherManager.h (hunk 3 above)
Comment (chrisdecenzo) [reply examples/tv-app/linux/include/content-launcher/ContentLauncherManager.h:54]: yes, that will be added to this PR

File: src/app/clusters/content-launch-server/content-launch-server.cpp (hunk 2 above)
Comment (amitnj) [reply src/app/clusters/content-launch-server/content-launch-server.cpp:149]: I see. But that code currently assumes that every content app supports the same set of clusters/commands/attributes. We can modify that later.

**Review by amitnj — APPROVED** [review]

**Review by Byungjoo-Lee — APPROVED** [review]

**Review by hawk248 — APPROVED** [review]

**Pull request merged**